packages/smithy-core/tests/functional/test_retries.py

```diff
@@ -0,0 +1,133 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+from asyncio import gather, sleep
+
+import pytest
+from smithy_core.exceptions import CallError, RetryError
+from smithy_core.interfaces import retries as retries_interface
+from smithy_core.retries import (
+    ExponentialBackoffJitterType,
+    ExponentialRetryBackoffStrategy,
+    StandardRetryQuota,
+    StandardRetryStrategy,
+)
+
+
+async def retry_operation(
+    strategy: retries_interface.RetryStrategy,
+    status_codes: list[int],
```

Comment:
Something I'm not sure if we're covering in this or the smithy-http client is what happens if we get an actual network issue like a timeout exception. Do we have code that's exercising that part of our retry process?